esmvalcore/preprocessor/_volume.py

```diff
@@ -295,6 +295,68 @@ def volume_statistics(
     return _create_cube_time(src_cube, result, times)
 
 
+def extract_surface(cube):
+    """
+    Extact the surface layer from a 3D cube.
+
+    Extracts the entire layer along the z axis where the z coordinate is
+    the minimum of the absolute value of the z-axis dimensions points.
+
+    Requires a cube with a z axis.
+
+    This assumes that the surface is the closest layer to zero.
+    This assumption is usually true in the ocean, but may not be the
+    case in all models for all z-axes.
+
+    In the case of temporally or spatially varying depth grid,
+    (ie the depth array is 3D or 4D), the surface layer is determined
+    using the mean of the depth points along the time, latitude and
+    longitude axes. This preprocessor may also behave strangely
+    if the z axis data are not monotonic, or unusual in some other way.
+
+    The preprocessor extract_layer can also do this, but it may be
+    much slower and more memory intensie as it regrids.
+    It is more robust to variability in the grid definitions.
+
+    Arguments
+    ---------
+    cube: iris.cube.Cube
+        input cube.
+
+    Returns
+    -------
+    iris.cube.Cube
+        collapsed cube.
+
+    """
+    # Get the z axis.
+    zcoord = cube.coord(axis='Z')
+    positive = zcoord.attributes['positive']
+
+    # Get the Z points:
+    if zcoord.points.ndim == 1:
+        points = zcoord.points
+    elif zcoord.points.ndim == 3:
+        points = zcoord.points.mean(axis=[1, 2])
+    elif zcoord.points.ndim == 4:
+        points = zcoord.points.mean(axis=[0, 2, 3])
```

Comment:
could `zcoord.points.ndim == 2`?

Reply:
either way, maybe add an exception if any of the dims are not the ones you analyze

Reply:
You're right. 

The problem with that (and this method as a whole) is that it makes assumptions about the z-axis position. For instance, what if you have a cube with dimensions (time, depth, latitude) ie a meridional transect? The preprocessor would fail.

Maybe instead  of that, we could replace this if-else loop with something more generic? 
```
    zcoord_dim = cube.coord_dims(zcoord)[0]
    axes = np.arrange(zcoord.points.ndim).pop(zcoord_dim)
    points = zcoord.points.mean(axis=axes)
```